brave/src/main/java/brave/internal/Platform.java

```diff
@@ -45,31 +45,30 @@
     }
   }
 
-  public Endpoint endpoint() {
+  @Nullable public abstract String getHostString(InetSocketAddress socket);
```

Comment:
Is the `String` suffix necessary? Isn't `getHostAddress` better since the type is string already?

Reply:
this one is a language level guard on InetSocketAddress.getHostAddress, which isn't available until v1.7. So `getHostAddress(socket)` reads similarly to `socket.getHostAddress` intentionally. I'll add javadoc as yeah not clear.